tests/Integration/Behaviour/Features/Context/Domain/Product/CommonProductFeatureContext.php

```diff
@@ -365,4 +363,26 @@ private function assertIndexation(string $productReference, bool $expectedIsInde
             )
         );
     }
+
+    private function assertProductType(string $productReference, string $productTypeName, int $shopId): void
+    {
+        $editableProduct = $this->getProductForEditing($productReference, $shopId);
+        Assert::assertEquals(
+            $productTypeName,
+            $editableProduct->getType(),
+            sprintf(
+                'Product type is not as expected. Expected %s but got %s instead',
+                $productTypeName,
+                $editableProduct->getType()
+            )
+        );
+        $productId = $this->getSharedStorage()->get($productReference);
+        $product = new Product($productId);
+        Assert::assertEquals($productTypeName === ProductType::TYPE_VIRTUAL, (bool) $product->is_virtual);
+        // cache_is_pack is automatically updated by legacy code when removing all pack items so it's not worth testing it for now
+        // Assert::assertEquals($productTypeName === ProductType::TYPE_PACK, (bool) $product->cache_is_pack);
```

Comment:
does it produce any error or what?
I don't understand why you say its not worth it?

Reply:
I didn't say anything I just moved some code 😅
I'll try if it works if I uncomment this line

Reply:
But I think it means we can't be sure that cache_is_ack will remain true as long as the product_type is Pack, because it is modified on product deletion and can be inconsistent sometimes, but I'm not sure fixing this is the PR's scope

Reply:
could you clarify the comment once you check the behavior? :pray: